core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -227,8 +227,13 @@ public void run() {
             killComputer(c);
         }
         getQueue().scheduleMaintenance();
-        for (ComputerListener cl : ComputerListener.all())
-            cl.onConfigurationChange();
+        for (ComputerListener cl : ComputerListener.all()) {
+            try {
+                cl.onConfigurationChange();
+            } catch (Throwable t) {
```

Comment:
Maybe exception? Catching `Error`s is considered as a bad idea since they should be fatal. But it's rather a holywar topic

Reply:
I thought about this too but have no strong opinion. FWIW there is similar code in [SaveableListener#fireOnChange](https://github.com/jenkinsci/jenkins/blob/d79c6afd2163a30b549314c7c1185bf8bd1fd149/core/src/main/java/hudson/model/listeners/SaveableListener.java#L78-L88) which handles `ThreadDeath` specially, not sure if there was a particular reason for that in that case.

Reply:
Yeah, we use Exception/Throwable catches randomly